lmcache/experimental/storage_backend/local_disk_backend.py

```diff
@@ -0,0 +1,159 @@
+
+logger = init_logger(__name__)
+
+
+@_lmcache_nvtx_annotate
+@torch.inference_mode()
+def save_disk(
+    path: str,
+    kv_chunk: torch.Tensor,
+    fmt_str: str,
+) -> None:
+    """
+    Save KV to disk.
+    """
+    save_file({"kv_chunk": kv_chunk}, path, {"fmt": fmt_str})
+    del kv_chunk
+
+
+@_lmcache_nvtx_annotate
+@torch.inference_mode()
+def load_disk(
+    path: str,
+    dst_device: str,
+) -> BufferMemoryObj:
+    """
+    Load KV from disk.
+    """
+    with safe_open(path, framework="pt",
+                   device=dst_device) as f:  # type: ignore
+        kv_chunk = f.get_tensor("kv_chunk")
+        metadata = f.metadata()
+    return BufferMemoryObj(
+        kv_chunk, BufferMemoryObjMetadata(MemoryFormat(int(metadata["fmt"]))))
+
+
+class LocalDiskBackend(StorageBackendInterface):
+
+    def __init__(self, config: LMCacheEngineConfig, dst_device: str = "cuda"):
+        self.dict: OrderedDict[CacheEngineKey,
+                               DiskCacheMetadata] = OrderedDict()
+        self.dst_device = dst_device
+
+        # TODO: Tune the number of workers for better performance
+        self.proc_pool_executor = ProcessPoolExecutor(max_workers=4)
+
+        self.disk_lock = threading.Lock()
+        assert config.local_disk is not None
+        self.path: str = config.local_disk
+        if not os.path.exists(self.path):
+            os.makedirs(self.path)
+            logger.info(f"Created local disk cache directory: {self.path}")
+
+        # TODO(Jiayi): Size and evictor should be configured
+
+        self.is_shutdown = False
+
+    def __str__(self):
+        return self.__class__.__name__
+
+    def contains(self, key: CacheEngineKey) -> bool:
+        with self.disk_lock:
+            return key in self.dict
+
+    def _key_to_path(
+        self,
+        key: CacheEngineKey,
+    ) -> str:
+        return self.path + key.to_string().replace("/", "-") + ".pt"
+
+    def insert_key(self, key: CacheEngineKey, size: int):
+        path = self._key_to_path(key)
+        with self.disk_lock:
+            self.dict[key] = DiskCacheMetadata(path, size)
+
+    def submit_put_task(
+        self,
+        key: CacheEngineKey,
+        memory_obj: MemoryObj,
+    ) -> Future:
+        kv_chunk = memory_obj.tensor
+        fmt_str = str(memory_obj.metadata.fmt.value)
+        path = self._key_to_path(key)
+        future = self.proc_pool_executor.submit(save_disk, path, kv_chunk,
+                                                fmt_str)
+        return future
+
+    def put_blocking(
+        self,
+        key: CacheEngineKey,
+        memory_obj: MemoryObj,
+    ) -> None:
+        """
+        Blocking put function.
+        This is for debugging and testing purposes.
+        """
+        path = self._key_to_path(key)
+        kv_chunk = memory_obj.tensor
+        fmt_str = str(memory_obj.metadata.fmt.value)
+        save_disk(path, kv_chunk, fmt_str)
+        self.insert_key(key, memory_obj.get_size())
```

Comment:
Seems like this function is not being called? Can we delete it?

Reply:
It's very help for debugging. We can drop it once the refactored version becomes stable